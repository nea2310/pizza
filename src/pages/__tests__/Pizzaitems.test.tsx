import { BrowserRouter } from 'react-router-dom';
import PizzaItems from '../pizzaItems/pizzaItems';
import { screen} from '@testing-library/react'
import { renderWithProviders } from '../../shared/testUtils/testUtils';
import { initialStore } from '../../shared/testUtils/mockedStore';

describe('PizzaItems page rendering', () => {
  it('Renders PizzaItems page', async() => {
      const favoritesPage = renderWithProviders(<BrowserRouter><PizzaItems /></BrowserRouter>, {
        preloadedState: initialStore
      });

      expect(favoritesPage).toMatchSnapshot();
  });
});
