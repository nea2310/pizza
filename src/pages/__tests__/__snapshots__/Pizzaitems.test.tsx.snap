// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PizzaItems page rendering Renders PizzaItems page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="pizza-items"
      >
        <h1
          class="pizza-items__main-header"
        >
          Заказать пиццу
        </h1>
        <div
          class="pizza-items__filter "
        >
          <button
            aria-label="фильтр поиска"
            class="pizza-items__filter-button"
          />
          <div
            class="pizza-items__filter-ingredients"
          >
            <fieldset
              class="ingredients-list"
            >
              <legend
                class="ingredients-list__legend"
              >
                Ингредиент
              </legend>
              <label
                class="ingredients-list__item"
              >
                <input
                  class="ingredients-list__checkbox"
                  name="моцарелла"
                  tabindex="-1"
                  type="checkbox"
                />
                <span
                  class="ingredients-list__checkmark"
                />
                моцарелла
              </label>
              <label
                class="ingredients-list__item"
              >
                <input
                  class="ingredients-list__checkbox"
                  name="пармезан"
                  tabindex="-1"
                  type="checkbox"
                />
                <span
                  class="ingredients-list__checkmark"
                />
                пармезан
              </label>
              <label
                class="ingredients-list__item"
              >
                <input
                  class="ingredients-list__checkbox"
                  name="горгонзола"
                  tabindex="-1"
                  type="checkbox"
                />
                <span
                  class="ingredients-list__checkmark"
                />
                горгонзола
              </label>
              <label
                class="ingredients-list__item"
              >
                <input
                  class="ingredients-list__checkbox"
                  name="орегано"
                  tabindex="-1"
                  type="checkbox"
                />
                <span
                  class="ingredients-list__checkmark"
                />
                орегано
              </label>
              <label
                class="ingredients-list__item"
              >
                <input
                  class="ingredients-list__checkbox"
                  name="томат"
                  tabindex="-1"
                  type="checkbox"
                />
                <span
                  class="ingredients-list__checkmark"
                />
                томат
              </label>
            </fieldset>
          </div>
          <div
            class="pizza-items__filter-spicy"
          >
            <fieldset
              class="radio"
            >
              <legend
                class="radio__legend"
              >
                Острая
              </legend>
              <div
                class="radio__group"
              >
                <label
                  class="radio__item"
                >
                  <input
                    class="radio__radiobutton"
                    name="spicy"
                    tabindex="-1"
                    type="radio"
                    value="да"
                  />
                  <span
                    class="radio__radiomark"
                  />
                  да
                </label>
                <label
                  class="radio__item"
                >
                  <input
                    class="radio__radiobutton"
                    name="spicy"
                    tabindex="-1"
                    type="radio"
                    value="нет"
                  />
                  <span
                    class="radio__radiomark"
                  />
                  нет
                </label>
                <label
                  class="radio__item"
                >
                  <input
                    class="radio__radiobutton"
                    name="spicy"
                    tabindex="-1"
                    type="radio"
                    value="не важно"
                  />
                  <span
                    class="radio__radiomark"
                  />
                  не важно
                </label>
              </div>
            </fieldset>
          </div>
          <div
            class="pizza-items__filter-lent"
          >
            <fieldset
              class="radio"
            >
              <legend
                class="radio__legend"
              >
                Можно в пост
              </legend>
              <div
                class="radio__group"
              >
                <label
                  class="radio__item"
                >
                  <input
                    class="radio__radiobutton"
                    name="lent"
                    tabindex="-1"
                    type="radio"
                    value="да"
                  />
                  <span
                    class="radio__radiomark"
                  />
                  да
                </label>
                <label
                  class="radio__item"
                >
                  <input
                    class="radio__radiobutton"
                    name="lent"
                    tabindex="-1"
                    type="radio"
                    value="нет"
                  />
                  <span
                    class="radio__radiomark"
                  />
                  нет
                </label>
                <label
                  class="radio__item"
                >
                  <input
                    class="radio__radiobutton"
                    name="lent"
                    tabindex="-1"
                    type="radio"
                    value="не важно"
                  />
                  <span
                    class="radio__radiomark"
                  />
                  не важно
                </label>
              </div>
            </fieldset>
          </div>
        </div>
        <ul
          class="pizza-items__product-cards"
        >
          <li
            class="pizza-items__product-card"
          >
            <article
              class="product-card"
            >
              <div
                class="product-card__favorites-button"
              >
                <button
                  aria-label="добавить в избранное"
                  class="favorites-button"
                />
              </div>
              <div
                class="product-card__content"
              >
                <h2
                  class="product-card__header"
                >
                  <a
                    class="product-card__link"
                    href="/product/fourCheesePizzaID"
                  >
                    Четыре сыра
                  </a>
                </h2>
                <img
                  alt="Четыре сыра"
                  class="product-card__img "
                  src="https://firebasestorage.googleapis.com/fourCheesePizza/"
                />
              </div>
              <div
                class="product-card__bottom"
              >
                <div
                  class="product-card__info"
                >
                  <span
                    class="product-card__price"
                  >
                    Цена: 
                    400
                    р.
                  </span>
                </div>
                <div
                  class="product-card__order-button"
                >
                  <button
                    class="order-button order-button_type_order"
                  >
                    Заказать
                  </button>
                </div>
              </div>
            </article>
          </li>
        </ul>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="pizza-items"
    >
      <h1
        class="pizza-items__main-header"
      >
        Заказать пиццу
      </h1>
      <div
        class="pizza-items__filter "
      >
        <button
          aria-label="фильтр поиска"
          class="pizza-items__filter-button"
        />
        <div
          class="pizza-items__filter-ingredients"
        >
          <fieldset
            class="ingredients-list"
          >
            <legend
              class="ingredients-list__legend"
            >
              Ингредиент
            </legend>
            <label
              class="ingredients-list__item"
            >
              <input
                class="ingredients-list__checkbox"
                name="моцарелла"
                tabindex="-1"
                type="checkbox"
              />
              <span
                class="ingredients-list__checkmark"
              />
              моцарелла
            </label>
            <label
              class="ingredients-list__item"
            >
              <input
                class="ingredients-list__checkbox"
                name="пармезан"
                tabindex="-1"
                type="checkbox"
              />
              <span
                class="ingredients-list__checkmark"
              />
              пармезан
            </label>
            <label
              class="ingredients-list__item"
            >
              <input
                class="ingredients-list__checkbox"
                name="горгонзола"
                tabindex="-1"
                type="checkbox"
              />
              <span
                class="ingredients-list__checkmark"
              />
              горгонзола
            </label>
            <label
              class="ingredients-list__item"
            >
              <input
                class="ingredients-list__checkbox"
                name="орегано"
                tabindex="-1"
                type="checkbox"
              />
              <span
                class="ingredients-list__checkmark"
              />
              орегано
            </label>
            <label
              class="ingredients-list__item"
            >
              <input
                class="ingredients-list__checkbox"
                name="томат"
                tabindex="-1"
                type="checkbox"
              />
              <span
                class="ingredients-list__checkmark"
              />
              томат
            </label>
          </fieldset>
        </div>
        <div
          class="pizza-items__filter-spicy"
        >
          <fieldset
            class="radio"
          >
            <legend
              class="radio__legend"
            >
              Острая
            </legend>
            <div
              class="radio__group"
            >
              <label
                class="radio__item"
              >
                <input
                  class="radio__radiobutton"
                  name="spicy"
                  tabindex="-1"
                  type="radio"
                  value="да"
                />
                <span
                  class="radio__radiomark"
                />
                да
              </label>
              <label
                class="radio__item"
              >
                <input
                  class="radio__radiobutton"
                  name="spicy"
                  tabindex="-1"
                  type="radio"
                  value="нет"
                />
                <span
                  class="radio__radiomark"
                />
                нет
              </label>
              <label
                class="radio__item"
              >
                <input
                  class="radio__radiobutton"
                  name="spicy"
                  tabindex="-1"
                  type="radio"
                  value="не важно"
                />
                <span
                  class="radio__radiomark"
                />
                не важно
              </label>
            </div>
          </fieldset>
        </div>
        <div
          class="pizza-items__filter-lent"
        >
          <fieldset
            class="radio"
          >
            <legend
              class="radio__legend"
            >
              Можно в пост
            </legend>
            <div
              class="radio__group"
            >
              <label
                class="radio__item"
              >
                <input
                  class="radio__radiobutton"
                  name="lent"
                  tabindex="-1"
                  type="radio"
                  value="да"
                />
                <span
                  class="radio__radiomark"
                />
                да
              </label>
              <label
                class="radio__item"
              >
                <input
                  class="radio__radiobutton"
                  name="lent"
                  tabindex="-1"
                  type="radio"
                  value="нет"
                />
                <span
                  class="radio__radiomark"
                />
                нет
              </label>
              <label
                class="radio__item"
              >
                <input
                  class="radio__radiobutton"
                  name="lent"
                  tabindex="-1"
                  type="radio"
                  value="не важно"
                />
                <span
                  class="radio__radiomark"
                />
                не важно
              </label>
            </div>
          </fieldset>
        </div>
      </div>
      <ul
        class="pizza-items__product-cards"
      >
        <li
          class="pizza-items__product-card"
        >
          <article
            class="product-card"
          >
            <div
              class="product-card__favorites-button"
            >
              <button
                aria-label="добавить в избранное"
                class="favorites-button"
              />
            </div>
            <div
              class="product-card__content"
            >
              <h2
                class="product-card__header"
              >
                <a
                  class="product-card__link"
                  href="/product/fourCheesePizzaID"
                >
                  Четыре сыра
                </a>
              </h2>
              <img
                alt="Четыре сыра"
                class="product-card__img "
                src="https://firebasestorage.googleapis.com/fourCheesePizza/"
              />
            </div>
            <div
              class="product-card__bottom"
            >
              <div
                class="product-card__info"
              >
                <span
                  class="product-card__price"
                >
                  Цена: 
                  400
                  р.
                </span>
              </div>
              <div
                class="product-card__order-button"
              >
                <button
                  class="order-button order-button_type_order"
                >
                  Заказать
                </button>
              </div>
            </div>
          </article>
        </li>
      </ul>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
